config/config.go

```diff
@@ -392,18 +333,23 @@ func (c *Configuration) Remotes() []string {
 	return c.remotes
 }
 
-// GitProtocol returns the protocol for the LFS API when converting from a
-// git:// remote url.
 func (c *Configuration) GitProtocol() string {
-	if value, ok := c.Git.Get("lfs.gitprotocol"); ok {
-		return value
+	return c.endpointConfig().GitProtocol()
+}
+
+func (c *Configuration) endpointConfig() *endpoint.Config {
```

Comment:
I really dig this. It's too bad that we have to keep something like it around in `*config.Configuration`, but it's nice that we can track all of the uses of it by checking out the callers of this function.